hanjaelee/app.js

```diff
@@ -52,20 +53,175 @@ app.post('/signup', (req, res) => {
       profile_image,
       password
     ) VALUES (?, ?, ?, ?);`
-  const rawData = database.query(rawQuery, [name, email, profileImage, password])
 
-  rawData
-    .then((data) => {
-      if (data.affectedRows) {
-        res.status(201).json({ message: 'userCreated' })
-        return
-      }
+  try {
+    const rawData = await database.query(rawQuery, [name, email, profileImage, password])
+    if (rawData['affectedRows']) {
+      return res.status(201).json({ message: 'userCreated' })
+    }
+    return res.status(202).json({ message: 'userNotCreated' })
+  } catch (error) {
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
+
+// 과제 3 게시글 등록하기
+app.post('/post', async (req, res) => {
+  const { title, content, userId } = req.body
+
+  const rawQuery =
+    `
+    INSERT INTO posts (
+      title,
+      content,
+      user_id
+    ) VALUES (?, ?, ?);`
+
+  try {
+    const rawData = await database.query(rawQuery, [title, content, userId])
+    if (rawData['affectedRows']) {
+      return res.status(201).json({ message: 'postCreated' })
+    }
+    return res.status(202).json({ message: 'postnotcreated' })
+  } catch (error) {
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
 
-      res.status(400).json({ message: 'userIsNotCreated' })
+// 과제 4 모든 게시물 가져오기
+app.get('/posts', async (req, res) => {
+  const rawQuery = `
+    SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      p.id AS postingId,
+      p.image_url AS postingImageUrl,
+      p.content AS postingContent
+    FROM posts AS p
+    INNER JOIN users AS u
+    ON p.user_id = u.id
+    `
+  try {
+    await database.query(rawQuery, (err, rows) => {
+      const result = {}
+      const data = rows
+      result['data'] = data
+      res.status(200).json(result)
     })
-    .catch((error) => {
-      res.status(400).json({ message: error.sqlMessage })
+  } catch (error) {
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
+
+// 과제 5 유저 한명이 작성한 모든 게시물 가져오기
+app.get('/posts/:userId', async (req, res) => {
+  const { userId } = req.params
+  const rawQuery = `
+    SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      JSON_ARRAYAGG(JSON_OBJECT(
+                    'postingId', p.id,
+                    'postingImageUrl', p.image_url,
+                    'postingContent', p.content)) AS posting
+    FROM posts AS p
+    INNER JOIN users AS u
+    ON p.user_id = u.id WHERE u.id = ${userId}
+    GROUP BY u.id;
+    `
+
+  try {
+    await database.query(rawQuery, (err, rows) => {
+      return res.status(200).json({ data: rows })
```

Comment:
해당 부분도 await 키워드를 사용하기 때문에 쿼리에 대한 결과값을 바로 변수에 할당에서 응답 data로 보내면 될 것 같습니다.

```suggestion
    const [result] = await database.query(rawQuery, (err, rows) => {
      return res.status(200).json({ data: result })
```